seed/seedingScript.js

```diff
@@ -1,14 +1,14 @@
 const fs = require('fs');
 const path = require('path');
 const { generateCSV, generateJSON } = require('./generateData');
-const { createHeaderImageRecord, createMainImageRecords } = require('./Page.js');
+const { createHeaderImageRecord, createMainImageRecords, createCouchRecord } = require('./Page.js');
 const filePath = path.join(__dirname, '/data');
 
 try {
   const headerImagesStream = fs.createWriteStream(filePath + '/headerImages.csv');
   generateCSV(headerImagesStream, createHeaderImageRecord, 1e7, () => {
```

Comment:
this sounds like a handy package, the function takes in the stream, a data generation function, presumably a number of times to write and a callback, right?

Reply:
i wrote this function myself. you can find it in `seed/generateData.js`. and yes, those are exactly the parameters 😄  i'm pretty proud if this one!

```
const fs = require('fs');
const path = require('path');

const timer = (record, start) => {
  if (record % 100000 === 0) {
    const now = Date.now();
    const minutesElapsed = (((now - start) / 1000) / 60);
    console.log(`created record #${record} - minutes elapsed: ${minutesElapsed}`);
  }
}

const start = Date.now();

module.exports.generateCSV = (writer, data, records, cb) => {
  let i = 0;
  write();
  function write() {
    let ok = true;
    do {
      i++;
      if (i === records) {
        writer.write(data(i), cb);
      } else {
        ok = writer.write(data(i));
        timer(i, start);
      }
    } while (i < records && ok);
    if (i < records) {
      writer.once('drain', write);
    }
  }
};
```